### PR TITLE
add shady link regex for check url with ips

### PR DESCRIPTION
**Pull request by sairuss7**
resolve https://github.com/NodeSecure/js-x-ray/pull/260

### CODE REVIEW

**Comment by PierreDemailly** [comment]
Hi @sairuss7 thanks for contributing 👍 

Could not we use simpler regexp? Maybe with the addition of a new constant `kShadyLinkExcludeRegExpp` to exclude local IPs (if needed, just an idea)

Moreover please add unit tests (`test/probes/isLiteral.spec.js`) and update the docs which only refer to suspicious extensions, we should add that it checks for IPs domains.

**Comment by jean-michelet** [comment]
> Could not we use simpler regexp?

Developing high-quality regexps requires a great amount of rigor, so I think we should refer to an official recommendation.

**Comment by sairuss7** [comment]
@PierreDemailly  Thanks for the feedback Pierre okay I'll do it better

**Comment by fraxken** [comment]
RegExp seems ok to me, but we should add new tests on the probe at least

File: src/probes/isLiteral.js [hunk 1]
```diff
@@ -4,13 +4,21 @@ import { builtinModules } from "repl";
 // Import Third-party Dependencies
 import { Hex } from "@nodesecure/sec-literal";
 
+
+const kMapRegexIps = Object.freeze({
+  regexIPv4With: /^(https?:\/\/)(?!127\.)(?!.*:(?:0{1,3}|25[6-9])\.)(?!.*:(?:25[6-9])\.(?:0{1,3}|25[6-9])\.)(?!.*:(?:25[6-9])\.(?:25[6-9])\.(?:0{1,3}|25[6-9])\.)(?!.*:(?:25[6-9])\.(?:25[6-9])\.(?:25[6-9])\.(?:0{1,3}|25[6-9]))((?:\d{1,2}|1\d{2}|2[0-4]\d|25[0-5])\.){3}(?:\d{1,2}|1\d{2}|2[0-4]\d|25[0-5])(?::\d{1,5})?(\/[^\s]*)?$/,
+  regexIPv6With: /^(https?:\/\/)(\[[0-9A-Fa-f:]+\])(?::\d{1,5})?(\/[^\s]*)?$/
+})
+
```
Comment (fraxken) [inline src/probes/isLiteral.js]:
```suggestion
const kMapRegexIps = Object.freeze({
  regexIPv4With: /^(https?:\/\/)(?!127\.)(?!.*:(?:0{1,3}|25[6-9])\.)(?!.*:(?:25[6-9])\.(?:0{1,3}|25[6-9])\.)(?!.*:(?:25[6-9])\.(?:25[6-9])\.(?:0{1,3}|25[6-9])\.)(?!.*:(?:25[6-9])\.(?:25[6-9])\.(?:25[6-9])\.(?:0{1,3}|25[6-9]))((?:\d{1,2}|1\d{2}|2[0-4]\d|25[0-5])\.){3}(?:\d{1,2}|1\d{2}|2[0-4]\d|25[0-5])(?::\d{1,5})?(\/[^\s]*)?$/,
  regexIPv6With: /^(https?:\/\/)(\[[0-9A-Fa-f:]+\])(?::\d{1,5})?(\/[^\s]*)?$/
});
```

**Review by fraxken — APPROVED** [review]

**Review by PierreDemailly — APPROVED** [review]

**Review by fraxken — APPROVED** [review]

**Comment by fraxken** [comment]
@allcontributors please add @sairuss7 for code

**Pull request merged**